### ISSUE TITLE
CASE operator works incorrectly

### ISSUE THREAD

**Issue by dimkas78** [ClickHouse/ClickHouse#38666]
ClickHouse version 22.3.3.44

When I use CASE operator with redundant section ELSE, it works as expected,
but if section ELSE is absent, operator always return only NULL value.

```
select 
    case 1 
        when 1 then 'a' 
        else 'b'
    end value
```
```
value
-----
a    
```
```
select 
    case 1 
        when 1 then 'a' 
    end value
```
```
value
-----
<NULL>      
```

**Comment by den-crane** [comment]
duplicate https://github.com/ClickHouse/ClickHouse/issues/2655 https://github.com/ClickHouse/ClickHouse/issues/9596

**Comment by alexey-milovidov** [comment]
Fixed in 23.5.
